app/models/name.rb

```diff
@@ -408,33 +457,181 @@ class Name < AbstractModel
     end
   end
 
-  scope :with_rank,
-        ->(rank) { where(rank: Name.ranks[rank]) if rank }
-
+  scope :of_lichens, -> { where(Name[:lifeform].matches("%lichen%")) }
+  scope :not_lichens, -> { where(Name[:lifeform].does_not_match("% lichen %")) }
+  scope :deprecated, -> { where(deprecated: true) }
```

Comment:
@nimmolo & @pellaea: How do you feel about moving the scopes to a different file in order to reduce file size. (The comments listing the scopes could remain here.)

Reply:
I tried moving the scopes out. It didn't work in a module, because the way files autoload under Zeitwerk, it just autoloads and throws an error on the word `scope` because the subfile isn't itself a Model. 

I wasn't sure how to include it otherwise. Feel free to try.

Reply:
@nimmolo: Thanks very much for the responses.
I'll make sure I have a better understanding of `Name.with_description_in_project(x)` and only change its name if I can find something better.
Thanks also for the info about Zeitwerk and moving scopes to its own file,

Reply:
@nimmolo 
It would help me if I knew where these scopes came from.
Some clearly reflect a filename in `app/classes/query` and/or 
a parameter in app/classes/pattern_search/name.rb or observation.rb

But I can't tell the origin of `Name.with_description_in_project`. 
The scope itself is broken:
```ruby
3.0.4 :002 > Name.with_description_in_project(266).first
  Name Load (0.4ms)  SELECT `names`.* FROM `names` INNER JOIN `name_descriptions` ON `name_descriptions`.`name_id` = `names`.`id` WHERE `names`.`project` = 266 ORDER BY `names`.`id` ASC LIMIT 1                                                   
/home/vagrant/.rvm/gems/ruby-3.0.4/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': Mysql2::Error: Unknown column 'names.project' in 'where clause' (ActiveRecord::StatementInvalid)                                                         
/home/vagrant/.rvm/gems/ruby-3.0.4/gems/mysql2-0.5.4/lib/mysql2/client.rb:148:in `_query': Unknown column 'names.project' in 'where clause' (Mysql2::Error)                                                                                         
```
I can make it work while returning reasonable results. 
But I can't be sure they are the right results without being sure of the intended use of the scope.

Maybe we should get rid of the scope until we have a need for it.

Reply:
My two cents again -- I think "with_descriptions_for_project" might be slightly easier to understand.  I think the name "with_description_in_project" might have been my name originally, but in retrospect -- if we can change any of these names willy nilly however we please -- I prefer "for" instead of "in".

Hmmm.  But I just checked, and it doesn't actually look like this flavor is used at all.  Maybe it would be better simply to remove it in honor of our continuing crusade to simplify MO wherever possible.  I support any of the above, though.